base/src/main/java/org/aya/concrete/parse/AyaProducer.java

```diff
@@ -615,25 +619,17 @@ private static SourcePos sourcePosForSubExpr(@NotNull SourceFile sourceFile, Seq
     return visitAtomPatterns(ctx.atomPatterns()).apply(true, null);
   }
 
-
   public BiFunction<Boolean, LocalVar, Pattern> visitAtomPatterns(@NotNull AyaParser.AtomPatternsContext ctx) {
     var atoms = ctx.atomPattern().stream()
       .map(this::visitAtomPattern)
       .collect(ImmutableSeq.factory());
-    if (atoms.sizeEquals(1)) return (ex, as) -> atoms.first().apply(ex);
+    if (atoms.sizeEquals(1)) return (ex, as) -> newBinOPScope(atoms.first().apply(ex), as);
```

Comment:
What's the point of one-arg binopseq?

Reply:
I see now. It's for section parsing.

Reply:
Although we do not want to support sections in patterns, we may. Do it in some day.

Reply:
It's to keep consistent with expr binopseq since they share the same name and the same pattern of parsing.
We may treat pattern section as underline pattern (like `x:` means `x:_`) (just random thoughts)